src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,526 @@
+        self.sites = sites
+        self.streams = acq_streams
+
+        self._log_path = log_path
+        if log_path:
+            filename = os.path.basename(self._log_path)
+            if not filename:
+                raise ValueError("Filename is not found on log path.")
+            self._exporter = find_fittest_converter(filename)
+            self._fn_bs, self._fn_ext = dataio.splitext(filename)
+            self._log_dir = os.path.dirname(self._log_path)
+
+        # internal class requirements between functions
+        self._pass_duration = []  # one value per milling setting, duration(s) after which drift acquisition starts
+        self._iterations = []  # one value per milling setting, numbers of scans of .roi
+        stage_md = stage.getMetadata()
+        try:
+            self._ion_beam_angle = stage_md[model.MD_ION_BEAM_TO_SAMPLE_ANGLE]
+        except KeyError:
+            raise ValueError("Ion beam angle not defined in stage metadata")
+        self._move_speed = MOVE_SPEED_DEFAULT  # stage speed used from _tiledacq.py
+
+        # estimate milling time from the list of milling settings-> millings
+        time_estimate = 0
+        drift_acq_time = 0
+        stage_time = 0
+
+        for setting_nb, setting in enumerate(self._settings):
+            time_estimate += setting.duration.value
+            if setting.dcRoi.value != UNDEFINED_ROI:
+                self._iterations.append(int(numpy.ceil(setting.duration.value / setting.dcPeriod.value)))
+                self._pass_duration.append(setting.duration.value / self._iterations[setting_nb])
+
+                drift_acq_time += self.estimate_drift_time(setting)
+            else:
+                self._iterations.append(1)
+                self._pass_duration.append(setting.duration.value)
+
+        prev_stage_pos_ref = None  # previous stage position used for reference in estimating time
+
+        for site in self.sites:
+            stage_time += self.estimate_stage_movement_time(site, prev_stage_pos_ref)
+            prev_stage_pos_ref = site.pos.value
+
+        self._milling_time_estimate = time_estimate + drift_acq_time  # time_estimate per feature
+        self._remaining_stage_time = stage_time  # total stage movement time
+
+    def cancel(self, future):
+        """
+        Canceler of acquisition task.
+        """
+        logging.debug("Canceling milling procedure...")
+
+        with future._task_lock:
+            if future._task_state == FINISHED:
+                return False
+            future._task_state = CANCELLED
+            future.running_subf.cancel()
+            logging.debug("Milling procedure cancelled.")  # add line number
+        return True
+
+    def estimate_milling_time(self, site_idx: int = 0, actual_time_per_site: float = None) -> float:
+        """
+        Estimates the milling time for the given feature.
+        :param site_idx: (int) index of the site to be milled
```

Comment:
"index of the site" is badly worded. Yes, that sort of is true, if you call this before you actually mill the site, and you index the sites from 0. I'd suggest to just describe as:
```suggestion
        :param sites_done: (int) number of sites already milled
```